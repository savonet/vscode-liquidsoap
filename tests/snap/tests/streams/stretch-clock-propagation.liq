# Stretch (as well as soundtouch) uses the Child_support clock control
# mechanism. We want to make sure that it properly propagates end-of-tracks
# to its underlying outputs/sources.

s = once(blank(duration=10.))

def on_stop() =
  test.pass()
end

output.dummy(fallible=true, on_stop=on_stop, s)

s = stretch(ratio=2., s)

clock.assign_new(sync="none",[s])

output.dummy(fallible=true, s)
