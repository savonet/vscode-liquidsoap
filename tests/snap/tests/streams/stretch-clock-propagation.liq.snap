># Stretch (as well as soundtouch) uses the Child_support clock control
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># mechanism. We want to make sure that it properly propagates end-of-tracks
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># to its underlying outputs/sources.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
>s = once(blank(duration=10.))
#^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                        ^^ source.liquidsoap constant.numeric.integer
#                          ^^^^ source.liquidsoap
>
>def on_stop() =
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^^^^^^^ source.liquidsoap
>  test.pass()
#^^^^^^^^^^^^^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>output.dummy(fallible=true, on_stop=on_stop, s)
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>
>s = stretch(ratio=2., s)
#^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                  ^ source.liquidsoap constant.numeric.integer
#                   ^^^^^^ source.liquidsoap
>
>clock.assign_new(sync="none",[s])
#^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                      ^ source.liquidsoap string.quoted.double.liquidsoap
#                       ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                           ^ source.liquidsoap string.quoted.double.liquidsoap
#                            ^^^^^^ source.liquidsoap
>
>output.dummy(fallible=true, s)
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>